generate/generate.go

```diff
@@ -388,6 +388,13 @@ func Generate(config *Config) (map[string][]byte, error) {
 		}
 	}
 
+	if g.Config.Optional == "generic" {
+		_, err = g.ref(g.Config.OptionalGenericType)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
`OptionalGenericType` must be defined if `optional: "generic"` is set, is this the appropriate place for a validation error to be thrown if it's unset?